Placeeo/Controller/MainViewController.swift

```diff
@@ -9,29 +9,30 @@ import UIKit
 import CoreLocation
 import MapKit
 
-class MainViewController: UIViewController, MKMapViewDelegate{
-    
+class MainViewController: UIViewController, MKMapViewDelegate, CLLocationManagerDelegate {
     private let mapView: MapView = {
         MapView.loadFromNib() as! MapView
     }()
     
-    let places: [Place] = [
-        Place(name: "Castle Uzhgorod", lat: 48.62216095313416, long: 22.308029690972464),
-        Place(name: "Church", lat: 48.623380761219515, long: 22.302193204377218)
-    ]
-    
     override func viewDidLoad() {
         super.viewDidLoad()
         view.addSubview(mapView)
-        
         NSLayoutConstraint.activate([
             mapView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
-            mapView.trailingAnchor.constraint(equalTo: view.leadingAnchor),
+            mapView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
             mapView.topAnchor.constraint(equalTo: view.topAnchor),
             mapView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
         ])
-        
-        mapView.show(places: places)
+        title = "Home"
+        LocationManager.shared.getUserLocation { [weak self] location in
+            DispatchQueue.main.async {
+                guard let strongSelf = self else { return }
```

Comment:
```suggestion
                guard let self = self else { return }
```
although, self is not used within the async block, but inside the getPlaces completion below. let's move [weak self] there